libraries/standard/json/src/json.c

```diff
@@ -160,9 +160,17 @@ static bool_ skipUTF8MultiByte( const char * buf,
                                 size_t max )
 {
     bool_ ret = false;
-    size_t i = *start, bitCount, j;
+    size_t i, bitCount, j;
     uint32_t value = 0;
-    uint8_t c = ( uint8_t ) buf[ i ];
+    uint8_t c;
+
+    assert( ( buf != NULL ) && ( start != NULL ) && ( max > 0U ) );
+
+    i = *start;
+    assert( i < max );
+    assert( buf[ i ] < 0 );
+
+    c = ( 0x80U | ( buf[ i ] & 0x7F ) );
```

Comment:
Magic numbers. Can these be made into named constants?

Reply:
These are bit masks.  They're actually easier to read as hex (as opposed to something like LOW_ORDER_SEVEN_BITS_HIGH - ick).

Reply:
Fair.